### ISSUE TITLE
I hate to suggest it, but adding splitting columns

### ISSUE THREAD

**Issue by wrightaprilm** [datacarpentry/python-ecology-lesson#245]
Hi all,

I was teaching a fork of these materials in a genomics class, and I realized something we don't cover is how to split a column into multiple columns. You can find an example about 5/6 of the way down the page [here](https://paleantology.github.io/SELUGandT/18-starting-with-data/), in which we've loaded some output from a Python package for post-processing some genomic data. The file has one column in which there are 5 variables in that column, separated by spaces. 

I suspect more than one variable per column is a fairly common data problem, and something people need to fix fairly often. This is covered in the OpenRefine lesson, but I don't think how to incorporate this type of cleaning is in either of the programmatic data processing lessons. 

With a Bug BBQ being scheduled, I figured this might be the time to raise the issue. For discussion:

1. Is this a broadly useful enough task to consider adding?
2. If so, where? For my purposes, I put it in starting with data because some of the more useful variables were mashed together in a column, and so to do anything fun, I needed to pull them apart. But I could see an argument for putting it in Combining Data, since there is a separate CSV file that contains this error. So combining dataframes would mean rectifying this error and standardizing the columns across frames.
3. Other thoughts?

**Comment by trallard** [comment]
Hi

This is a task that I _often_ encounter when dealing with data. I would be a bit reluctant to add this too early in the lessons (e.g starting with data). It makes a bit more sense to have this added later on as people start getting more familiar with data manipulations in pandas.

**Comment by wrightaprilm** [comment]
Yeah. The "Putting It All Together" lesson needs substantial work. Maybe it could go in there, so that lesson is less of a straight rehash, and has some novel content.

**Comment by stijnvanhoey** [comment]
I agree that this is a common operation. I like the suggestion of adapting the content of the "Putting It All Together" lesson to incorporate this. Maybe change it to explain some of the [`.str.`-functions](https://pandas.pydata.org/pandas-docs/stable/text.html) and [`datetime`-functions](https://pandas.pydata.org/pandas-docs/stable/timeseries.html) of Pandas, which would enable to incorporate the `str.split` operation: e.g. from split to two columns:
```
df[["first", "last"]] = df["combination_column"].str.split("_", expand=True)
```

To not overload the session, I would suggest to remove thecurrent matplotlib `plt.`-section as it is actually new content which feels (just my feeling) a bit off in that episode. Furthermore, the object oriented way of matplotlib (using `fig` and `ax` objects to manipulate) could be more powerful, because it interacts easy with Pandas and can bring a kind of `best of both worlds`:
```
fig, ax = plt.subplots() #prepare a matplotlib figure

flowdata.plot(ax=ax) # use pandas for the plotting

# Provide further adaptations with matplotlib:
ax.set_xlabel("")
ax.tick_params(labelsize=15, pad=8)
fig.suptitle('Flow station time series', fontsize=15)
```
Hence, by introducing it as such, it connects with the quick plot function of episode 1 and illustrates how to combine with matplotlib. 
(sorry if I'm going off-topic here, I can maybe create a separate issue?)

**Comment by stijnvanhoey** [comment]
@wrightaprilm, no problem. Unfortunately, I won't be able to join during the bug BBQ. Maybe the discussion above can result in a more general adaptation of the `07-putting-it-all-together` episode. 

Points to consider:
1. adding a section on the `.str.**` methods, with the splitting as application? 
2. also introduce the `.dt.**` functions for datetimes (it is a similar idea as 1)?
3. what about `matplotlib`, useful to consider in that episode? Maybe just exclude that part, but if yes, consider the object oriented way of development to improve the connection with earlier lessons. Maybe useful with respect to the discussion in #246: You can, just as with pandas plot outputs, further adapt `plotnine` plots using matplotlib. The trick is to use the `draw()` function to access the `plotnine` output as a matplotlib object.
